plugins/kotlin/src/main/java/org/quiltmc/quiltflower/kotlin/struct/KType.java

```diff
@@ -0,0 +1,124 @@
+package org.quiltmc.quiltflower.kotlin.struct;
+
+import kotlin.reflect.jvm.internal.impl.metadata.ProtoBuf;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.java.decompiler.struct.gen.VarType;
+import org.jetbrains.java.decompiler.util.TextBuffer;
+import org.quiltmc.quiltflower.kotlin.metadata.MetadataNameResolver;
+import org.quiltmc.quiltflower.kotlin.util.KTypes;
+
+public class KType {
```

Comment:
Also needs comments here. Also, is it possible to extend VarType? I would rather the type system be more uniform. See GenericType for more information.